### PR TITLE
Fix RTH hover above home

### PR DESCRIPTION
**Pull request by shellixyz**
The state machine was switching to NAV_STATE_RTH_LANDING even when landing was not allowed which made the throttle during hover unable to go over cruise throttle impeding the ability of the craft to stay at the target altitude.

~Don't merge yet I need to test this fix.~

### CODE REVIEW

**Comment by shellixyz** [comment]
Tested on FW, RTH loiter around home is now working as intended. Ready to merge.

**Comment by digitalentity** [comment]
Maybe it's better to fix HOVER_PRIOR_TO_LANDING state?

**Comment by shellixyz** [comment]
@digitalentity I chose this way for these reasons:
1. It would have meant duplicating the sanity checks 
2. It would have made the code harder to read in my opinion
3. I added NAV_RC_ALT to the final hover and it wouldn't have made sense to add it to the `HOVER_PRIOR_TO_LANDING` state

**Comment by digitalentity** [comment]
@shellixyz good point, agreed

**Review by digitalentity — APPROVED** [review]

File: src/main/io/osd.c [hunk 1]
```diff
@@ -678,13 +678,12 @@ static const char * navigationStateMessage(void)
...
-                    return OSD_MESSAGE_STR("LOITERING AROUND HOME");
-                }
-                return OSD_MESSAGE_STR("HOVERING");
-            }
             return OSD_MESSAGE_STR("LANDING");
+        case MW_NAV_STATE_HOVER_ABOVE_HOME:
+            if (STATE(FIXED_WING))
+                return OSD_MESSAGE_STR("LOITERING AROUND HOME");
+            else
+                return OSD_MESSAGE_STR("HOVERING");
```
Comment (fiam) [inline src/main/io/osd.c]: Could you drop the else and add braces in the if?

Comment (shellixyz) [reply src/main/io/osd.c]: Drop the else ? Why ?

Comment (fiam) [reply src/main/io/osd.c]: Drop the else, not its contents :-). Something like...
```
+if (STATE(FIXED_WING)) {
+    return OSD_MESSAGE_STR("LOITERING AROUND HOME");
+}
+return OSD_MESSAGE_STR("HOVERING");
```

Comment (shellixyz) [reply src/main/io/osd.c]: Oh ok :+1:. Done

Comment (fiam) [reply src/main/io/osd.c]: Thanks!

**Pull request merged**